llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -509,6 +509,8 @@ void SelectionDAGISel::initializeAnalysisResults(
     FnVarLocs = &FAM.getResult<DebugAssignmentTrackingAnalysis>(Fn);
 
   auto *UA = FAM.getCachedResult<UniformityInfoAnalysis>(Fn);
+  MMI = &MFAM.getResult<MachineModuleAnalysis>(*MF).getMMI();
```

Comment:
IIUC it should be obtained from `ModuleAnalysisManager` or `ModuleAnalysisManagerMachineFunctionProxy`, but current `ModuleAnalysisManagerMachineFunctionProxy` is an `OuterAnalysisManagerProxy`. 

off-topic: Ideally, it should be thread-safe to obtain module analysis results in machine function pass pipeline.

Reply:
I was expecting this to not compile if it wasn't going to work. I'm not sure how to get a ModuleAnalysisManager from the MachineFunctionAnalysisManager

Reply:
I think the registration is just not happening and I'm not sure why. I can't run the analysis standalone with -passes either

Reply:
ping. Is MachineModuleInfo just broken in new PM?

Reply:
```suggestion
  MMI = MFAM.getResult<ModuleAnalysisManagerMachineFunctionProxy>(*MF)
            .getCachedResult<MachineModuleAnalysis>(*MF->getFunction().getParent());
```